examples/htool_console.c

```diff
@@ -244,6 +244,40 @@ static int set_uart_config(struct libhoth_usb_device *dev,
 
 int htool_console_run(struct libhoth_usb_device *dev,
                       const struct htool_console_opts *opts) {
+  if (opts->snapshot) {
+    printf("Reading console snapshot\n");
+    // 1. Create Debug interface (should be already created by this time.)
+    // 2. read console snapshot
+    const size_t kMailboxSize = 1024;
+    size_t response_bytes_written;
+    int status = htool_exec_hostcmd(dev, EC_CMD_CONSOLE_REQUEST,
+      0, NULL, 0, NULL, 0, &response_bytes_written);
+    if (status != LIBHOTH_OK) {
+      printf("EC_CMD_CONSOLE_REQUEST not OK.\n");
+      return status;
+    }
+
+    struct ec_params_console_read_v1 read_request = { .subcmd = CONSOLE_READ_NEXT };
+    const size_t max_bytes_per_read = kMailboxSize - sizeof(struct ec_host_response);
+    char ret[12288] = { 0 };
+    int idx = 0;
+    while (true) {
+      char buf[kMailboxSize];
+      status = htool_exec_hostcmd(
+        dev, EC_CMD_CONSOLE_READ,
+        0, &read_request, sizeof(read_request), buf, max_bytes_per_read, &response_bytes_written);
+      if (status != LIBHOTH_OK) {
+        printf("EC_CMD_CONSOLE_READ not OK.\n");
+        return status;
+      }
+      strcat(ret, buf);
```

Comment:
`strcat` will gleefully write past the end of the buffer, and there's no guarantee that the firmware only has 0x3000 bytes to provide.  At least change this to `strncat(ret, buf, sizeof(buf));`.

Reply:
Done.